components/scream/src/share/util/scream_tridiag.hpp

```diff
@@ -244,6 +253,35 @@ void thomas_amxm (DT* const dl, DT* d, DT* const du, XT* X,
   }
 }
 
+template <typename TridiagDiag>
+KOKKOS_INLINE_FUNCTION
+void bfb_thomas_factorize (TridiagDiag dl, TridiagDiag d, TridiagDiag du,
+                           typename std::enable_if<TridiagDiag::rank == 1>::type* = 0) {
+  const int nrow = d.extent_int(0);
+  assert(dl.extent_int(0) == nrow);
```

Comment:
Should we be using scream_kassert instead of raw asserts?

Reply:
I can if there's a preference. Tbc, `assert` is supported on device: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#assertion, so I believe `scream_kassert` and `assert` are functionally identical. I'm trying to keep scream_tridiag.hpp pretty separate from scream infrastructure b/c I need to copy-paste it over to HOMME for the NH conversion effort until the projects can intermingle dependencies. Right now I have the dependencies down to just `util::min`.

Reply:
>  I'm trying to keep scream_tridiag.hpp pretty separate from scream infrastructure b/c I need to copy-paste it over to HOMME for the NH conversion

OK, that's a good reason to not use our scream macros.